flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -450,7 +450,10 @@ private void resetSynchronousSavepointId(long id, boolean succeeded) {
             activeSyncSavepointId = null;
             operatorChain.setIgnoreEndOfInput(false);
         }
-        syncSavepointId = null;
+
+        if (syncSavepointId != null && syncSavepointId == id) {
```

Comment:
How about we change it to?: 
```suggestion
        if (syncSavepointId != null && syncSavepointId <= id) {
```